### ISSUE TITLE
Enable data_source uid output to address grafana 3.x dashboard changes

### ISSUE THREAD

**Issue by fluential** [grafana/terraform-provider-grafana#304]
## Enable `data_source` `uid` output to address grafana 3.x dashboard changes
In the output of Grafana 3.x dashboard json data source reference has changed:

Was
```json
"datasource": "<name-of-datasource>"
```

It is now:
```json
"datasource" : {
  "type": "cloudwatch"
  "uid" "<unique_identifier>"
}
```
Terraform grafana provider does not expose in any way `uid`. This breaks dashboard provisioning from json templates. As a suggestion having an output with this field available for reference would be useful.

**Comment by wanieldilson** [comment]
Essential for us, our whole dashboard provisioning workflow is now dead

**Comment by wanieldilson** [comment]
So I looked into this, I tried adding the necessary code in the provider but got errors so had to open [this PR](https://github.com/grafana/grafana-api-golang-client/pull/42) to pull the uid details through.

Once the above is merged I will open a new PR against the provider to allow `uid` as an attribute that you can reference.

At least that's the plan anyway...

**Comment by wanieldilson** [comment]
[This PR](https://github.com/grafana/terraform-provider-grafana/pull/315) is now open and will hopefully close this issue once merged.  🤞
